### PR TITLE
[TGUI] GPS

### PR DESCRIPTION
**Pull request by dearmochi**
## What Does This PR Do

- Ports GPS from Browser UI to TGUI
  - The UI now auto-refreshes as a result
  - The UI now displays distance and direction to another signal
- Adds ability to only display signals on the same Z-level
- Cleans up, refactors and documents the GPS code

## Why It's Good For The Game

TGUI good, GPS less of a pain to use

## Images of changes

image

## Changelog
:cl:
add: GPS interface has been remade and improved in TGUI
/:cl:

### CODE REVIEW

**Comment by Carthusia** [comment]
yes! No more having to reuse it to update it!

File: tgui/packages/tgui/interfaces/GPS.js [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+            ml="0.25rem"
+            onClick={() => act('tag', { newtag: newTag })}>
+            <Icon name="pen" />
+          </Button>
+        </LabeledList.Item>
+        <LabeledList.Item label="Range">
+          <Button
+            selected={!same_z}
+            icon={same_z ? "compress" : "expand"}
+            content={same_z ? "Normal" : "Extended"}
```
Comment (AffectedArc07) [inline tgui/packages/tgui/interfaces/GPS.js:140]: `Normal` and `Extended` dont seem right here. Maybe `Local Sector` and `Global`?

**Comment by AffectedArc07** [comment]
I also feel like this UI could be wider
image

This is quite squished

**Comment by Kiyahitayika** [comment]
Absolutely love this!

**Review by AffectedArc07 — APPROVED** [review]

**Comment by hal9000PR** [comment]
you should go 122m North East. UI looks great, great quality of life improvement.

**Pull request merged**